### ISSUE TITLE
Proposal: new `esmax` option; deprecate `esnext` and `es3`

### ISSUE THREAD

**Issue by jmeas** [jshint/jshint#2256]
@gnarf and I discussed the current state of the es flags in jshint, and came up with a system that could be simpler, and more flexible, than the current options.

Presently, there are two options to manage what features of different ECMAScript versions jshint will allow: `es3` and `esnext`. `es3` was preceded by an `es5` flag.

`esnext` is best thought of as `es6`, as it doesn't seem to support potential es7 features. @brianloveswords mentioned that he had been told that there's no plan to add es7 features to the `esnext` flag for some time.

What all of these flags have in common is that they specify the **maximum** ECMAScript version that you support. `es3: true` means "I only support features up to ES3." It is incompatible with `esnext`, which currently means "I only support features up to ES6."

To write it out:

##### `esmax` - The maximum version of ECMAScript that your library supports

**Accepts**: Integers (3, 5, 6, 7)
**Default**: `5`

===

@tkellen weighed in on providing per-feature flags that could further tweak these options. For instance, something like:

```js
{
  esmax: 7,
  esfeatures: {
    exponentiationOperator: false
  }
}
```

There might be better solutions than the one outlined here, but I thought I'd get the discussion started.

**Comment by jugglinmike** [comment]
I proposed the language version option a few months ago (albeit with a different name) in gh-2124. Thanks to the goal of One JavaScript, it's generally safe to say that "ES3 code is ES6 code", so given a language version, additional semantics like "max" or "support features up to" are unnecessary. I'd like to close this as a duplicate of that issue as it seems like they effectively describe the same thing. If I'm missing something (or you feel strongly about the name `esmax`), we can continue the discussion there. Sound good?

As for options for non-standardized language features, those really belong under a suitably-named option. In gh-1979, [I proposed `experimental`](https://github.com/jshint/jshint/pull/1979#issuecomment-63234755), but now I think `unstable` would be much better. In any case, this should probably be tracked with a dedicated issue.
